hedera-node/src/test/java/com/hedera/services/txns/crypto/CryptoDeleteTransitionLogicTest.java

```diff
@@ -119,111 +139,106 @@ void acceptsValidTxn() {
-
-		// when:
-		subject.doStateTransition();
-
+		given(accountStore.loadAccount(any())).willThrow(new InvalidTransactionException(ACCOUNT_DELETED));
 		// then:
-		verify(txnCtx).setStatus(ACCOUNT_DELETED);
+		assertFailsWith(() -> subject.doStateTransition(), ACCOUNT_DELETED);
 	}
 
 	@Test
 	void followsHappyPath() {
+		// given:
 		givenValidTxnCtx();
-
+		given(accountStore.loadAccount(Id.fromGrpcAccount(targetId))).willReturn(target);
+		given(accountStore.loadAccount(Id.fromGrpcAccount(beneficiaryId))).willReturn(beneficiary);
+		given(tokenStore.isKnownTreasury(Id.fromGrpcAccount(targetId))).willReturn(false);
+		given(target.isSmartContract()).willReturn(false);
+		given(beneficiary.isSmartContract()).willReturn(false);
+		given(copyOnWriteIds.getAsIds()).willReturn(List.of(IdUtils.asToken("1.2.3")));
+		given(target.getAssociatedTokens()).willReturn(copyOnWriteIds);
+		given(tokenStore.loadToken(any())).willReturn(token);
+		given(token.isDeleted()).willReturn(true);
 		// when:
 		subject.doStateTransition();
-
 		// then:
-		verify(ledger).delete(target, payer);
-		verify(txnCtx).setStatus(SUCCESS);
+		verify(accountStore, times(2)).persistAccount(any());
+		verify(recordService).includeHbarBalanceChanges(any());
 	}
 
 	@Test
 	void rejectsDetachedAccountAsTarget() {
-		// setup:
+		// given:
 		givenValidTxnCtx();
-		given(ledger.isDetached(target)).willReturn(true);
-
-		// when:
-		subject.doStateTransition();
-
-		// when:
-		verify(ledger, never()).delete(target, payer);
-		verify(txnCtx).setStatus(ACCOUNT_EXPIRED_AND_PENDING_REMOVAL);
+		given(accountStore.loadAccount(Id.fromGrpcAccount(targetId))).willReturn(target);
+		given(accountStore.loadAccount(Id.fromGrpcAccount(beneficiaryId))).willReturn(beneficiary);
+		given(tokenStore.isKnownTreasury(Id.fromGrpcAccount(targetId))).willReturn(false);
+		given(target.isSmartContract()).willReturn(true);
+		// then:
+		assertFailsWith(() -> subject.doStateTransition(), ACCOUNT_EXPIRED_AND_PENDING_REMOVAL);
 	}
 
 	@Test
 	void rejectsDetachedAccountAsReceiver() {
-		// setup:
-		var receiver = IdUtils.asAccount("0.0.7676");
-
-		givenValidTxnCtx(receiver);
-		given(ledger.isDetached(receiver)).willReturn(true);
-
-		// when:
-		subject.doStateTransition();
-
-		// when:
-		verify(ledger, never()).delete(target, receiver);
-		verify(txnCtx).setStatus(ACCOUNT_EXPIRED_AND_PENDING_REMOVAL);
+		givenValidTxnCtx();
+		given(accountStore.loadAccount(Id.fromGrpcAccount(targetId))).willReturn(target);
+		given(accountStore.loadAccount(Id.fromGrpcAccount(beneficiaryId))).willReturn(beneficiary);
+		given(tokenStore.isKnownTreasury(Id.fromGrpcAccount(targetId))).willReturn(false);
+		given(target.isSmartContract()).willReturn(false);
+		given(beneficiary.isSmartContract()).willReturn(true);
+		// then:
+		assertFailsWith(() -> subject.doStateTransition(), ACCOUNT_EXPIRED_AND_PENDING_REMOVAL);
 	}
 
+	//TODO: This passing test should probably be removed/
 	@Test
 	void capturesFailInvalid() {
```

Comment:
Instead of deleting the test, you can rename it to something meaningful and applicable to the test scenario - for example, `propagatesInvalidTransactionExceptionsProperly` . Feel free to think of a name yourself :)